app/assets/javascripts/application.js.erb

```diff
@@ -31,48 +31,56 @@ function human_time(seconds_total) {
   return `${hours_str}:${minutes_str}:${seconds_str}`;
 }
 
+function handleErrors(response) {
+    if (!response.ok) {
+        table.processing(false);
+        throw Error(response.statusText);
+    }
+    return response;
+}
+
+function checkResponse(type) {
+    if (Array.isArray(type)) {
+        table.rows.add(type).draw()
+    } else {
+        show_errors([type])
+    }
+}
+
 function fetch_table_data(table, options){
   if (!options) options = {};
   if (!options.doneCallback) options.doneCallback = null;
   if (!options.base_uri) options.base_uri = window.location.pathname;
 
-  oboe({
-    url: options.base_uri + '/jobs.json?'+get_request_params(),
+  fetch(options.base_uri + '/jobs.json?' + get_request_params(), {
     headers: {
-      'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
-      'X-Requested-With': 'XMLHttpRequest',
-      'Content-Type': 'application/json'
+        'X-CSRF-Token': $('meta[name="csrf-token"]').attr('content'),
+        'X-Requested-With': 'XMLHttpRequest',
+        'Content-Type': 'application/json'
     }
-  }).start(function(){
+  })
+  .then(handleErrors)
+  .then((response) => {
     table.processing(true);
-  }).node('data.*', function(jobs){
-    // data is an array of arrays, where each array has one or more jobs
-    // this is called once for each array of jobs
-    table.rows.add(jobs).draw();
-    table.processing(false);
-  }).node('errors.*', function(error){
-    // errors is a single array of error messages
-    // this is called once for each error msg
-    show_errors([error]);
-  }).done(function(){
-    table.processing(false);
-
-    if(options.doneCallback){
-       options.doneCallback();
-    }
-  }).fail(function(errorReport){
-    if(errorReport.statusCode != null){
-      show_errors(["Request for jobs failed with status code: " + errorReport.statusCode]);
-    }
-    else{
-      //FIXME: this error appears even when the above 404 occurs, for example
-      // that is because a 404 responce for json request returns a plain text response
-      // and parsing that as json fails
-       show_errors(["Request for jobs failed due to body parsing error."])
-    }
-
-    table.processing(false);
-  });
+    return ndjsonStream(response.body);
+  })
+  .then((stream) => {
+    const reader = stream.getReader();
+    let read;
+    reader.read().then(read = (result) => {
+        if (result.done) {
+            table.processing(false);
+            if (options.doneCallback) {
+                options.doneCallback();
+            }
+            return;
+        }
+        checkResponse(result.value);
+        reader.read().then( read );
```

Comment:
see comment above about inlining the checkResponse function. put newlines between the `if (result.done)` block, the checkResponse block, and the next read call. Add a comment above the `reader.read().then( read );`: "recurse through the stream"